### PR TITLE
Save previous map elements on resize

### PR DESCRIPTION
**Pull request by furdarius**
Map losing list elements on resize.
This PR fixes it with setting elements from old map to new map.

### CODE REVIEW

File: map.c [hunk 1]
```diff
@@ -101,8 +101,21 @@ map_resize(map_t *m, int new_cap)
     if (!nm) {
         return -1;
     }
-    memcpy(m, nm, sizeof(map_t));
-    map_free(nm);
+
+    for (int i = 0; i < m->len; i++) {
```
Comment (briandowns) [inline map.c:105]: Ugh... This is what I initially had (for the most part)(this impl is better) and I decided to use memcpy for some reason.

**Comment by briandowns** [comment]
Thank you very much for this!

**Comment by furdarius** [comment]
No problem :)

**Comment by briandowns** [comment]
Seems to be erroring with `make: *** [test] Abort trap: 6`. I assume you saw different behavior?

**Comment by briandowns** [comment]
Disregard. Small push made.

**Pull request merged**